src/frontend/src/optimizer/property/monotonicity.rs

```diff
@@ -271,3 +292,53 @@ impl MonotonicityAnalyzer {
         Inherent(Unknown)
     }
 }
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, Hash)]
+pub struct MonotonicityMap(BTreeMap<usize, Monotonicity>);
```

Comment:
Can you add comments? The meaning of the BTreeMap is not clear to me. Is it sth like `[Monotonicity; num_cols]`?